File: invenio_communities/members/services/permissions.py
# -*- coding: utf-8 -*-
#
# Copyright (C) 2022 Northwestern University.
#
# Invenio-Communities is free software; you can redistribute it and/or modify
# it under the terms of the MIT License; see LICENSE file for more details.

"""Members Permission Policy."""

from invenio_records_permissions.generators import AnyUser, SystemProcess
from invenio_records_permissions.policies import BasePermissionPolicy


class CommunityMembersPermissionPolicy(BasePermissionPolicy):
    """Permissions for Community Members CRUD operations."""

    # TODO #384
    can_read = [AnyUser()]

    can_search = [AnyUser(), SystemProcess()]

File: invenio_communities/assets/semantic-ui/js/invenio_communities/members/components/bulk_actions/permissions.js
/*
 * This file is part of Invenio.
 * Copyright (C) 2022 CERN.
 *
 * Invenio is free software; you can redistribute it and/or modify it
 * under the terms of the MIT License; see LICENSE file for more details.
 */

/*
 * To set a PermissionPolicy we need to do it in the following format:
 * {
 *  <permission_plicy>: <array_of_id>
 * }
 * if the permission policy that is defined is not part of the permissions
 * then we will filter only the values with the id we set in the array.
 */

export const RolePermissionPolicy = {
  can_invite_owners: ["manager", "curator", "reader"],
};

export const filterOptionsByPermissions = (options, policy, permissions) => {
  if (Object.keys(policy).length != 1) {
    throw 'PermissionPolicy object must contain only 1 pair "key, value"';
  }
  for (const [key, value] of Object.entries(policy)) {
    if (permissions[key]) {
      return options;
    } else {
      return options.filter(({ name }) => value.includes(name));
    }
  }
};
